### ISSUE TITLE
`prmd doc` incorrectly applies `$ref` links if properties are out of alphabetical order

### ISSUE THREAD

**Issue by duckpunch** [interagent/prmd#297]
Here is an example.  (I'm using YAML because I find it more readable.  I don't think it affects the problem here.)

**directory structure**

```bash
$ tree
.
├── schema.json
└── schemata
    ├── bar.yaml
    └── foo.yaml

1 directory, 3 files
```
**foo.yaml**

```yaml
---
$schema: http://json-schema.org/draft-04/hyper-schema

id: schemata/foo
title: foo

type: object
properties:
  # Note the fields are not in alphabetical order
  b_field:
    type: string
  a_field:
    $ref: /schemata/bar#/properties/bar_id
```

**bar.yaml**

```yaml
---
$schema: http://json-schema.org/draft-04/hyper-schema

id: schemata/bar
title: bar

type: object
properties:
  bar_id:
    type: string
```

**running prmd doc**

```bash
$ prmd combine schemata -o schema.json && prmd doc schema.json
...
### Attributes

| Name | Type | Description | Example |
| ------- | ------- | ------- | ------- |
| **a_field** | *string* |  | `"example"` |
| **[b_field](#resource-bar)** | *string* |  | `"example"` |
...
```

Note that `b_field` is linked instead of `a_field`.  The problem in the Markdown is fixed if I change the ordering in the schemata.  Also, `schema.json` is always "right".

**schema.json**

```json
...
      "properties": {
        "b_field": {
          "type": [
            "string"
          ]
        },
        "a_field": {
          "$ref": "#/definitions/bar/properties/bar_id"
        }
      }
...
```

**Comment by geemus** [comment]
Interesting, I had actually seen a couple cases of mis-linking similar to this, but never quite put my finger on how/why. The alpha sorting seems like a good lead, I'll try to dig in from there.

**Comment by geemus** [comment]
released in 0.11.11

**Comment by duckpunch** [comment]
Thanks for the quick turnaround!
